src/main/java/seedu/moneygowhere/parser/ConsoleParser.java

```diff
@@ -156,50 +255,29 @@ private static void validateCommandAddExpenseValues(CommandLine commandLine) thr
                 ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_NAME_LONG
         );
 
-        if (name.isBlank()) {
+        if (isNameArgumentInvalid(name)) {
             throw new ConsoleParserCommandAddExpenseInvalidException(
-                    Messages.CONSOLE_ERROR_COMMAND_ADD_EXPENSE_INVALID
+                    Messages.CONSOLE_ERROR_COMMAND_ADD_EXPENSE_ARG_NAME_INVALID
             );
         }
 
         String amountStr = commandLine.getOptionValue(
                 ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_AMOUNT_LONG
         );
 
-        if (amountStr == null) {
-            throw new ConsoleParserCommandAddExpenseInvalidException(
-                    Messages.CONSOLE_ERROR_COMMAND_ADD_EXPENSE_INVALID
-            );
-        }
-
-        try {
-            BigDecimal amount = new BigDecimal(amountStr);
-            if (amount.compareTo(BigDecimal.ZERO) != 1) {
-                throw new ConsoleParserCommandAddExpenseInvalidException(
-                        Messages.CONSOLE_ERROR_COMMAND_INVALID_AMOUNT
-                );
-            }
-        } catch (NumberFormatException exception) {
+        if (isAmountArgumentInvalid(amountStr)) {
             throw new ConsoleParserCommandAddExpenseInvalidException(
-                    Messages.CONSOLE_ERROR_COMMAND_ADD_EXPENSE_INVALID
+                    Messages.CONSOLE_ERROR_COMMAND_ADD_EXPENSE_ARG_AMOUNT_INVALID
             );
         }
 
-        String modeOfPayment = commandLine.getOptionValue(
-                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_MODE_OF_PAYMENT
+        String dateTimeStr = commandLine.getOptionValue(
+                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_DATE_TIME_LONG
         );
 
-        if (modeOfPayment != null
-                && !(modeOfPayment.equalsIgnoreCase(
-                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_MODE_OF_PAYMENT_VAL_CASH)
-                || modeOfPayment.equalsIgnoreCase(
-                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_MODE_OF_PAYMENT_VAL_PAYNOW)
-                || modeOfPayment.equalsIgnoreCase(
-                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_MODE_OF_PAYMENT_VAL_PAYLAH)
-                || modeOfPayment.equalsIgnoreCase(
-                ConsoleParserConfigurations.COMMAND_ADD_EXPENSE_ARG_MODE_OF_PAYMENT_VAL_CARD))) {
```

Comment:
Thanks for removing the modeOfPayment checks! But I think we need to change the syntax slightly for the UG as well as the corresponding error messages.

Reply:
Thanks for pointing it out. I'll make the change now.